src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -7952,7 +7969,46 @@ private BoundExpression MakeMemberAccessValue(BoundExpression expr, BindingDiagn
             }
         }
 
-        private BoundExpression BindInstanceMemberAccess(
+#nullable enable
+        // When we're binding a member access that is not invoked and the member lookup yielded no result:
+        // - if an extension member lookup finds a non-method extension member, then that's the member being accessed.
+        //
+        // - if the extension member lookup finds a method (classic extension method compatible with the receiver or method in extension declaration;
+        //   closer than any non-method extension member), then we return nothing and let the caller represent the failed member lookup with a BoundMethodGroup.
+        //   Note: Such method group will be resolved specially in scenarios that can handle method groups
+        //     (such as inferred local `var x = A.B;`, conversion to a delegate type `System.Action a = A.B;`).
+        //     It will be an error in other scenarios.
+        //
+        // - if the extension member lookup is ambiguous, then we'll use an error symbol as the result of the member access.
+        //
+        // - if the extension member lookup finds nothing, then we return nothing and let the caller represent the failed member lookup with a BoundMethodGroup.
+        internal BoundExpression? ResolveExtensionMemberAccessIfResultIsNonMethod(SyntaxNode syntax, BoundExpression receiver, string name,
+            SeparatedSyntaxList<TypeSyntax> typeArgumentsSyntax, ImmutableArray<TypeWithAnnotations> typeArgumentsOpt,
+            BindingDiagnosticBag diagnostics)
+        {
+            CompoundUseSiteInfo<AssemblySymbol> useSiteInfo = GetNewCompoundUseSiteInfo(diagnostics);
+            // Note: we're resolving without arguments, which means we're not treating the member access as invoked
+            var resolution = this.ResolveExtension(
+                syntax, name, analyzedArguments: null, receiver, typeArgumentsOpt, options: OverloadResolution.Options.None,
+                returnRefKind: default, returnType: null, withDependencies: useSiteInfo.AccumulatesDependencies);
+
+            diagnostics.Add(syntax, useSiteInfo); // PROTOTYPE test use-site info
+
+            if (resolution.IsNonMethodExtensionMember(out Symbol? extensionMember))
+            {
+                Debug.Assert(typeArgumentsSyntax.IsEmpty());
+                diagnostics.AddRange(resolution.Diagnostics); // PROTOTYPE test dependencies/diagnostics
+                resolution.Free();
+
+                return GetExtensionMemberAccess(syntax, receiver, extensionMember, diagnostics);
+            }
+
+            resolution.Free();
+            return null;
+        }
+#nullable disable
+
+        private BoundExpression BindMemberAccessWithBoundLeftCore(
```

Comment:
>BindMemberAccessWithBoundLeftCore [](http://example.com/codeflow?start=32&length=33)

It is not clear what motivated the rename. Especially that "Core" suffix isn't rely informative. Is there "non-Core" version? What does that do vs. this one doesn't?

Reply:
Was the "Instance" part of the original name inaccurate? I think we need to come up with a clearer, more descriptive name.

Reply:
From offline discussion, yes the issue was with "Instance". The bound left may not be an instance. I'll take a stab at improving the name

Reply:
I ended up reverting this. I still think the name isn't great, but it's not essential for our present purpose